### ISSUE TITLE
[Branch 2.4] Assign char * to size_t

### ISSUE THREAD

**Issue by bluecub** [babelouest/ulfius#68]
Hi @babelouest 
I found this in multiple  places

In orcania o_strstr returns char * as below
`char * o_strstr(const char * haystack, const char * needle);`

yet in u_websocket.c, in few places line 1055,1059 and others, you are assigning it to size_t as below
`response_offset = o_strstr(response + response_offset, "\r\n") + 2;`

Cheers,

**Comment by babelouest** [comment]
Hello @bluecub ,

In C, if you add a positive integer value to a pointer, you simply move forward in memory.

Assuming that you have the following:
```C
char * var = "Hello World!"; // var is an array of char
printf("var at index 0: %s\nvar at 6: %s\n", var, (var + 6));
```

This should print the following:
```shell
var at index 0: Hello World!
var at index 6: World!
```

In the case you mention, the code `o_strstr(response + response_offset, "\r\n")` means that it searches for the string `"\r\n"` in the string `response` but starting at the offset `response_offset`.

Although, this is a quick and dirty code to make the development easier but eventually it will be replaced by a more secure and more efficient code.

**Comment by bluecub** [comment]
Hi @babelouest, 
yeah I understand that, but It won't work on clang... I just gave it a try...

> /home/code/app/lib/ulfius/src/u_websocket.c:1055:35: error: incompatible pointer to integer conversion assigning to 'size_t' (aka 'unsigned long') from 'char *' [-Werror,-Wint-conversion]
>                   response_offset = o_strstr(response + response_offset, "\r\n") + 2;
>                                   ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /home/code/app/lib/ulfius/src/u_websocket.c:1059:35: error: incompatible pointer to integer conversion assigning to 'size_t' (aka 'unsigned long') from 'char *' [-Werror,-Wint-conversion]
>                   response_offset = o_strstr(response + response_offset, "\r\n") + 2;
>                                   ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /home/code/app/lib/ulfius/src/u_websocket.c:1063:35: error: incompatible pointer to integer conversion assigning to 'size_t' (aka 'unsigned long') from 'char *' [-Werror,-Wint-conversion]
>                   response_offset = o_strstr(response + response_offset, "\r\n") + 2;
>                                   ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 

Haven't tried build 2.4 on FreeBSD but I hope the new u_websocket.c won't cause problems...
Thanks for everything

**Comment by babelouest** [comment]
My bad,

I didn't understand the question at first, I was in another dimension...

This bug is fixed now in the 2.4 branch, but this code will certainly have massive change until the release.

Thanks @bluecub !
